lib/options_validator.js

```diff
@@ -94,6 +81,61 @@ function validate(
 
   const verifiedOptions = Object.assign({}, providedOptions);
 
+  Object.keys(optionsSchema).forEach(optionName => {
```

Comment:
We should reintroduce the ability to opt out of validation now:
```
if (optionsValidationLevel === VALIDATION_LEVEL_NONE) {
  return verifiedOptions;
}
```

Reply:
Opting out of validation is done [here](https://github.com/mongodb/node-mongodb-native/pull/1884/files#diff-e1fa59f7e4e4c9080c24be96453dfdb8R99). We can't introduce opting out at the start of the iteration because we need to check required and deprecated options.